app/features/quizzify/core.py

```diff
@@ -6,31 +6,31 @@
 
 logger = setup_logger()
 
-def executor(files: list[ToolFile], topic: str, num_questions: int, verbose=False):
-    
+
+def executor(files, topic: str, num_questions: int, verbose=False):
```

Comment:
It's good practice to have a type annotation here. Perhaps since it is no longer a "ToolFile" we should implement an abstract interface that the files references for type safety